src/libraries/System.Net.Sockets/src/System/Net/Sockets/TCPListener.cs

```diff
@@ -118,7 +119,11 @@ public void AllowNatTraversal(bool allowed)
                 throw new InvalidOperationException(SR.net_tcplistener_mustbestopped);
             }
 
-            _serverSocket.SetIPProtectionLevel(allowed ? IPProtectionLevel.Unrestricted : IPProtectionLevel.EdgeRestricted);
+            if (_serverSocket != null)
+            {
+                SetIPProtectionLevel(allowed);
+            }
+            _allowNatTraversal = allowed;
```

Comment:
What's the expected behavior if the server is started, then stopped, and then started again?  It looks like previously the AllowNatTraversal setting would have only applied to the first socket, but here we're never clearing this value, so it'll also end up applying to the second socket?  I don't know if that's a good change or not.

Reply:
Hmm. That's a valid point. Most likely, the old behavior was unintentional, but the change can break something.

Reply:
I changed the logic to follow the current behavior on not propagating this option to a new socket and submitted a new issue #32653 to investigate the problem further.